discovery-infra/tests/base_test.py

```diff
@@ -178,7 +178,8 @@ def collect_test_logs(self, cluster, api_client, test, nodes):
         log_dir_name = f"{env_variables['log_folder']}/{test.name}"
         with suppress(ApiException):
             cluster_details = json.loads(json.dumps(cluster.get_details().to_dict(), sort_keys=True, default=str))
-            download_logs(api_client, cluster_details, log_dir_name, test.result_call.failed)
+            if test.result_call.failed:
```

Comment:
I think that this condition should be checked on the [cluster teardown](https://github.com/openshift/assisted-test-infra/blob/7ffdc1bf69463a9df01f3f765a1444175e1b0ba0/discovery-infra/tests/base_test.py#L76). no?

Reply:
right. this method is being called in cluster teardown
https://github.com/openshift/assisted-test-infra/blob/master/discovery-infra/tests/base_test.py#L76

Reply:
This method seems more like a utility. I think it would be much proper to make that check in the calling teardown function

Reply:
Fixed